packages/bpk-component-autosuggestV2/src/BpkAutosuggest.tsx

```diff
@@ -0,0 +1,406 @@
+} from 'downshift';
+
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import ClearIconLg from '../../bpk-component-icon/lg/close-circle';
+import { withButtonAlignment } from '../../bpk-component-icon';
+import BpkInput, { INPUT_TYPES } from '../../bpk-component-input';
+import { cssModules } from '../../bpk-react-utils';
+
+import STYLES from './BpkAutosuggest.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+const ClearButtonIcon = withButtonAlignment(ClearIconLg);
+
+export type BpkAutoSuggestTheme = {
+  container?: string;
+  containerOpen?: string;
+  suggestionsList?: string;
+  suggestion?: string;
+  suggestionHighlighted?: string;
+  sectionContainer?: string;
+  sectionTitle?: string;
+  clearButton?: string;
+  label?: string;
+  desktopLabel?: string;
+  input?: string;
+  inputWrapper?: string;
+  inputTextWrapper?: string;
+  desktopSuggestionsContainer?: string;
+  desktopSuggestionsList?: string;
+};
+
+export type BpkAutoSuggestProps<T> = {
+  ariaLabels: {
+    resultsList?: string;
+    label?: string;
+    clearButton?: string;
+  };
+  getSuggestionValue: (suggestion: T) => string;
+  id: string;
+  inputProps: HTMLProps<HTMLInputElement>;
+  onSuggestionsClearRequested: () => void;
+  onSuggestionSelected?: (data?: {
+    inputValue: string;
+    suggestion?: T;
+  }) => void;
+  onSuggestionsFetchRequested: (value: string) => void;
+  renderSuggestion: (suggestion: T) => ReactElement;
+  suggestions: T[];
+  getA11yResultsMessage: (resultCount: number) => string;
+  defaultValue?: string;
+  enterKeyHint?: string;
+  isBanana?: boolean;
+  isDesktop?: boolean;
+  onClick?: () => void;
+  onLoad?: (inputValue: string) => void;
+  renderBesideInput?: () => ReactElement;
+  showClear?: boolean;
+  theme?: Partial<BpkAutoSuggestTheme>;
+  highlightFirstSuggestion?: boolean;
+  shouldRenderSuggestions?: (value?: string) => boolean;
+  multiSection?: boolean;
+  getSectionSuggestions?: (section: T) => T[];
+  renderSectionTitle?: (section: T) => ReactElement;
+  alwaysRenderSuggestions?: boolean;
+  onInputValueChange?: (input: { method: string; newValue: string }) => void;
+};
+
+const defaultTheme = {
+  container: getClassName('bpk-autosuggest__container'),
+  containerOpen: getClassName('bpk-autosuggest__container--open'),
+  suggestionsContainer: getClassName('bpk-autosuggest__suggestions-container'),
+  suggestionsContainerOpen: getClassName(
+    'bpk-autosuggest__suggestions-container--open',
+  ),
+  suggestionsList: getClassName('bpk-autosuggest__suggestions-list'),
+  suggestion: getClassName('bpk-autosuggest__suggestion-item'),
+  suggestionHighlighted: getClassName(
+    'bpk-autosuggest__suggestion-item--highlighted',
+  ),
+  sectionContainer: getClassName('bpk-autosuggest__section-container'),
+  sectionTitle: getClassName('bpk-autosuggest__section-title'),
+};
+
+const BpkAutosuggest = forwardRef<HTMLInputElement, BpkAutoSuggestProps<any>>(
```

Comment:
Just to check as there's a lot of new props - were these all (except for those mentioned in the migration guide already) already supported by the `react-autosuggest` library?

Reply:
Yea. I went through all the main BpkAutosuggest implementations in skyscanner to find what props were used and passed through.

Slingshot uses alwaysRenderSuggestions - this keeps the list open when the input looses focus.

Carhire and hotels use highlightFirstSuggestion, shouldRenderSuggestions, multiSection and the getters associated

The banana needs are marked up in the description

Reply:
amazing! I don't like that some of our bpk components refer to an external lib's docs because that opens the API WAY too much